### ISSUE TITLE
Bug with set_readonly and symlinks

### ISSUE THREAD

**Issue by montyvesselinov** [JuliaLang/Pkg.jl#1474]
The following crashed:

```
rm -rf ~/.julia
git clone https://github.com/JuliaRegistries/General
cd General
git checkout registrator/mads/d6bdc55b/v0.8.0
git remote rm origin
```

Now, start a Julia 1.2.0 REPL:
```
julia> using Pkg
julia> pkg"registry add ."
julia> pkg"add Mads@0.8.0"
julia> using Mads
```

**Comment by fredrikekre** [comment]
https://github.com/JuliaRegistries/General/pull/4770#issuecomment-546883724

**Comment by fredrikekre** [comment]
Fixed by 90d83cc39acab69d12e162fb3d0638f8c66227b7

**Comment by montyvesselinov** [comment]
@fredrikekre is this fixed now in j1.3?! I tested with nightly and it was still failing?!
